lib/routes/whu/swrh.ts

```diff
@@ -0,0 +1,117 @@
+        },
+    ],
+    parameters: { type: '公告类型，详见表格' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '水利水电学院公告',
+    maintainers: ['FanofZY'],
+    handler,
+    description: `| 公告类型 | 学院新闻 | 学术科研 | 通知公告 |
+  | -------- | -------- | -------- | -------- |
+  | 参数     | 0        | 1        | 2        |`,
+};
+
+async function handler(ctx) {
+    const type = Number.parseInt(ctx.req.param('type'));
+
+    let link;
+    switch (type) {
+        case 0:
+            link = `${baseUrl}/index/xyxw.htm`; // 学院新闻
+            break;
+
+        case 1:
+            link = `${baseUrl}/index/xsky.htm`; // 学术科研
+            break;
+
+        case 2:
+            link = `${baseUrl}/xxgk/tzgg.htm`; // 通知公告
+            break;
+
+        default:
+            throw new Error(`Unknown type: ${type}`);
+    }
+
+    const response = await got(link);
+    const $ = load(response.data);
+
+    const list =
+        type === 0
+            ? $('div.my_box_nei')
+                  .toArray()
+                  .map((item) => {
+                      item = $(item);
+                      return {
+                          title: item.find('a b.am-text-truncate').text().trim(),
+                          pubDate: item.find('a i').text().trim(),
+                          link: new URL(item.find('a').attr('href'), baseUrl).href,
+                      };
+                  })
+            : $('div.list_txt.am-fr ul.am-list li')
+                  .toArray()
+                  .map((item) => {
+                      item = $(item);
+                      return {
+                          title: item.find('a span').text().trim(),
+                          pubDate: item.find('a i').text().trim(),
+                          link: new URL(item.find('a').attr('href'), baseUrl).href,
+                      };
+                  });
+
+    let items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                // 首先检查是否是微信公众号
+                item.description = item.link.includes('weixin')
+                    ? await fetchArticle(item.link).then((article) => article.description)
+                    : await (async () => {
+                          try {
+                              const response = await got(item.link);
+                              const $ = load(response.data);
+
+                              return $('.v_news_content').length ? $('.v_news_content').html().trim() : $('.prompt').length ? $('.prompt').html() : item.title;
+                          } catch {
+                              return item.title;
+                          }
+                      })();
+
+                item.pubDate = $('meta[name="PubDate"]').length ? timezone(parseDate($('meta[name="PubDate"]').attr('content')), +8) : item.pubDate;
```

Comment:
This condition is always false since `$` is referencing https://github.com/DIYgod/RSSHub/blob/cd38c282de105691f41db09ca61e97950780526f/lib/routes/whu/swrh.ts#L60-L61